src/System.IO.FileSystem/tests/FileStream/WriteByte.cs

```diff
@@ -59,5 +59,29 @@ public void WriteFile()
                 Assert.Equal(TestBuffer, buffer);
             }
         }
+
+        [Fact]
+        public void WriteFile_PastBufferSize()
+        {
+            string fileName = GetTestFilePath();
+            using (FileStream fs = new FileStream(fileName, FileMode.Create))
+            {
+                for (int written = 0; written < TestBuffer.Length * 1000; written++)
+                {
+                    fs.WriteByte(TestBuffer[written % TestBuffer.Length]);
+                }
+            }
+
+            using (FileStream fs = new FileStream(fileName, FileMode.Open))
+            {
+                byte[] buffer = new byte[TestBuffer.Length];
+                int offset = 0;
+                while (fs.Read(buffer, 0, TestBuffer.Length) > 0)
+                {
+                    offset += TestBuffer.Length;
+                    Assert.Equal(TestBuffer, buffer);
```

Comment:
I'd also suggest clearing the buffer after the assert, so that if Read were to fail but still return a positive value, the next assert would fail, rather than pass because it was examining the old contents.